packages/tpetra/core/ext/TpetraExt_MatrixMatrix_HIP.hpp

```diff
@@ -0,0 +1,781 @@
+// @HEADER
+// ***********************************************************************
+//
+//          Tpetra: Templated Linear Algebra Services Package
+//                 Copyright (2008) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+
+#ifndef TPETRA_MATRIXMATRIX_HIP_DEF_HPP
+#define TPETRA_MATRIXMATRIX_HIP_DEF_HPP
+
+#ifdef HAVE_TPETRA_INST_HIP
+namespace Tpetra {
+namespace MMdetails {
+
+/*********************************************************************************************************/
+// MMM KernelWrappers for Partial Specialization to HIP
```

Comment:
Can you say (in the comments) what changed from other specializations to accommodate HIP?
(I don't know this code well enough to spot differences or errors.)
Longer term (not this PR), it might be good to refactor this code and all its specializations; I noticed a lot of duplication when doing UVM removal.

Reply:
Nothing at all changed, it's just a copy-paste and then replace "KokkosCudaWrapperNode" with "KokkosHIPWrapperNode". It is a lot of duplication, but that's kind of unavoidable when the different backends are selected by template specialization. We can fix the duplication later with enable_if or making these specialized functions just thin wrappers over generic algorithms.